### PR TITLE
Update the Quantity Selector UI and Fix change quantity

### PR DESCRIPTION
**Pull request by vitoria**
#### What is the purpose of this pull request?
Update the quantity selector UI and Fix change quantity.

#### What problem is this solving?
The Mini Cart Item wasn't using the styleguide's Numeric Stepper and there was an error when changing the item quantity.

#### How should this be manually tested?
https://waza--storecomponents.myvtex.com/admin/cms/storefront

#### Screenshots or example usage
captura de tela de 2018-08-27 11-28-36

#### Types of changes
- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Requires change to documentation, which has been updated accordingly.

### CODE REVIEW

File: react/components/MiniCartContent.js [hunk 1]
```diff
@@ -39,6 +39,17 @@ class MiniCartContent extends Component {
     return sum
   }
 
+  getNoRepetedItems = () =>
```
Comment (claudivanfilho) [inline react/components/MiniCartContent.js]: I think that `getDistinctItems` or `getGroupedItems` is a better name to this function

File: react/components/MiniCartContent.js [hunk 2]
```diff
@@ -70,29 +81,72 @@ class MiniCartContent extends Component {
...
-      this.setState({
-        showSpinner: false,
+    const items = this.getNoRepetedItems()
+    const itemPayloadConcatenated = items.find(item => item.id === id)
+    const itemsPayload = orderForm.items.filter(item => item.id === id)
+    const index = orderForm.items.indexOf(itemsPayload[0])
+    const newQuantity = quantity - (itemPayloadConcatenated.quantity - itemsPayload[0].quantity)
+
+    if (newQuantity > 0) {
+      const updatedItem = [itemsPayload[0]].map(item => {
```
Comment (claudivanfilho) [inline react/components/MiniCartContent.js]: there is no need of this array here

File: react/components/MiniCartContent.js [hunk 3]
```diff
@@ -70,29 +81,72 @@ class MiniCartContent extends Component {
...
+          orderFormId: orderForm.orderFormId,
+          items: updatedItem,
+        },
+      }).then(() => {
+        this.setState({
+          showSpinner: false,
+        })
+      })
+    } else {
+      let updatedItem = [itemsPayload[0]].map(item => {
```
Comment (claudivanfilho) [inline react/components/MiniCartContent.js]: same here

File: react/components/MiniCartContent.js [hunk 4]
```diff
@@ -70,29 +81,72 @@ class MiniCartContent extends Component {
...
+      })
+
+      updateOrderForm({
+        variables: {
+          orderFormId: orderForm.orderFormId,
+          items: updatedItem,
+        },
+      })
+
+      updatedItem = [itemsPayload[1]].map(item => {
```
Comment (claudivanfilho) [inline react/components/MiniCartContent.js]: same here

File: react/components/MiniCartContent.js [hunk 5]
```diff
@@ -70,29 +81,72 @@ class MiniCartContent extends Component {
...
-        orderFormId: orderForm.orderFormId,
-        items: updatedItem,
-      },
-    }).then(() => {
-      this.setState({
-        showSpinner: false,
+    const items = this.getNoRepetedItems()
+    const itemPayloadConcatenated = items.find(item => item.id === id)
+    const itemsPayload = orderForm.items.filter(item => item.id === id)
+    const index = orderForm.items.indexOf(itemsPayload[0])
```
Comment (claudivanfilho) [inline react/components/MiniCartContent.js]: use `head` of ramda.

U r using `itemsPayload[0]` many times in the code, better to assign to a variable.

**Review by claudivanfilho — CHANGES_REQUESTED** [review]

**Review by salesfelipe — APPROVED** [review]

File: react/global.css [hunk 6]
```diff
@@ -1,11 +1,11 @@
 .vtex-minicart__item {
-  height: 109px;
+  height: 125px;
 }
 
 .vtex-minicart__img-container {
-  width: 50px;
-  height: 50px;
-  min-width: 50px;
+  width: 65px;
```
Comment (claudivanfilho) [inline react/global.css:6]: use w3 of tachyons

File: react/global.css [hunk 7]
```diff
@@ -1,11 +1,11 @@
...
+  height: 65px;
```
Comment (claudivanfilho) [inline react/global.css:7]: use h3 of tachyons

File: react/global.css [hunk 8]
```diff
@@ -124,13 +124,24 @@
   height: 100%;
 }
 
+.vtex-minicart__item-name {
+  height: 34px;
```
Comment (claudivanfilho) [inline react/global.css:128]: use h2 of tachyons

File: react/global.css [hunk 9]
```diff
@@ -124,13 +124,24 @@
...
   overflow: hidden;
   text-overflow: ellipsis;
   max-width: 90%;
 }
 
+.vtex-minicart__item-name-100 > .vtex-product-name > .vtex-product-name__brand {
+  white-space: nowrap;
+  overflow: hidden;
+  text-overflow: ellipsis;
+  max-width: 100%;
```
Comment (claudivanfilho) [inline react/global.css:142]: use `.mw-100` of tachyons

File: react/global.css [hunk 10]
```diff
@@ -124,13 +124,24 @@
...
+
 .vtex-minicart__item-name > .vtex-product-name > .vtex-product-name__brand {
   white-space: nowrap;
   overflow: hidden;
   text-overflow: ellipsis;
   max-width: 90%;
 }
 
+.vtex-minicart__item-name-100 > .vtex-product-name > .vtex-product-name__brand {
+  white-space: nowrap;
```
Comment (claudivanfilho) [inline react/global.css:139]: use `truncate` of tachyons

File: react/global.css [hunk 11]
```diff
@@ -124,13 +124,24 @@
...
+  overflow: hidden;
+  text-overflow: ellipsis;
+  max-width: 100%;
+}
+
 .vtex-minicart__item-sku > .vtex-product-name > .vtex-product-name__brand {
   white-space: nowrap;
```
Comment (claudivanfilho) [inline react/global.css:146]: Same here

**Review by claudivanfilho — CHANGES_REQUESTED** [review]

**Review by claudivanfilho — APPROVED** [review]

**Review by andreldsa — APPROVED** [review]

**Pull request merged**